runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java

```diff
@@ -88,18 +98,21 @@ public void onSetMessage(SetMessage message)
 
 		if (ALLOWED_HISTORY.contains(message.getType()))
 		{
-			final QueuedMessage queuedMessage = QueuedMessage.builder()
+			final QueuedMessageBuilder queuedMessageBuilder = QueuedMessage.builder()
 				.type(message.getType())
 				.name(message.getName())
 				.sender(message.getSender())
-				.value(nbsp(message.getValue()))
-				.runeLiteFormattedMessage(nbsp(message.getMessageNode().getRuneLiteFormatMessage()))
-				.build();
+				.value(nbsp(message.getValue()));
 
-			if (!messageQueue.contains(queuedMessage))
-			{
-				messageQueue.offer(queuedMessage);
-			}
+			// Check to see if this is a formatted message from RuneLite - such the GE/HA value on examine
+			if (message.getValue().contains("<col"))
```

Comment:
why was this added?

Reply:
This was added to ensure that colored messages (ie the extra message with the Grand Exchange & High Alch value when you examine an item) shows up colored when you hop or login. It wasn't maintaining the color style with the previous way the builder was setup. 

I'm not sure those should even be saved to the history, but they were showing up as the same chat message type, EXAMINE_ITEM, and showing up as all black text upon hopping, which looked off.

Reply:
I don't really understand this. The RuneLite formatted message is not in the same format as the normal messages, so I don't think this would work right.